### PR TITLE
Implement a basic auth microservice with tests

### LINKED ISSUE #1: Auth microservice
Authorize user via an OpenID provider specified in a configuration file and return JWT token

### PR DESCRIPTION
**Pull request by mmskv**
Closes #1

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "python.linting.pylintEnabled": true,
+    "python.linting.enabled": true,
+    "python.formatting.provider": "black"
+}
```
Comment (TheHussky) [inline .vscode/settings.json:5]: IMHO, this should be gitignored

File: auth/auth/main.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+oauth = OAuth(app)
+register_oauth_providers_from_config(oauth, app.config)
+
+
+@app.route("/login/<provider>")
+def login(provider):
+    if provider not in app.config["OAUTH_PROVIDERS"]:
+        return 501
+
+    redirect_uri = url_for(f"auth", provider=provider, _external=True)
```
Comment (TheHussky) [inline auth/auth/main.py:25]: Isn't `f"auth" == "auth"` ?

File: auth/config.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+GOOGLE_CLIENT_ID = os.getenv("GOOGLE_CLIENT_ID")
+GOOGLE_CLIENT_SECRET = os.getenv("GOOGLE_CLIENT_SECRET")
+GOOGLE_AUTHORIZE_PARAMS = {"hd": "miem.hse.ru"}  # Should not be used without validation of the returned ID token
+GOOGLE_METADATA_URL = "https://accounts.google.com/.well-known/openid-configuration"
+
+KEYCLOAK_CLIENT_ID = os.getenv("KEYCLOAK_CLIENT_ID")
+KEYCLOAK_CLIENT_SECRET = os.getenv("KEYCLOAK_CLIENT_SECRET")
+
+JWT_SECRET_KEY = os.getenv("JWT_SECRET_KEY")
```
Comment (TheHussky) [inline auth/config.py:19]: Don't you mean private key?

File: auth/config.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import os
+
+SECRET_KEY = os.getenv("FLASK_SECRET_KEY")
```
Comment (TheHussky) [inline auth/config.py:3]: Don't you mean private key?

**Review by TheHussky — COMMENTED** [review]
Lacks tests; 
Left some comments in places that require editing in my opinion

File: .vscode/settings.json (hunk 1 above)
Comment (mmskv) [reply .vscode/settings.json:5]: fair, I have ragequitted vscode in favor of nvim since then

File: auth/auth/main.py (hunk 2 above)
Comment (mmskv) [reply auth/auth/main.py:25]: fixed in upcoming commit

File: auth/config.py (hunk 3 above)
Comment (mmskv) [reply auth/config.py:19]: JWT doesn't support PKI and sharing RSA pubkeys sounds broken
So symmetric keys it is

File: auth/config.py (hunk 4 above)
Comment (mmskv) [reply auth/config.py:3]: google "flask secret key"

**Pull request merged**